pkg/transientstore/store_helper.go

```diff
@@ -0,0 +1,190 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package transientstore
+
+import (
+	"sort"
+	"sync"
+)
+
+// loadPvtRWSetMap is a loader function of store.cache
+func loadPvtRWSetMap(key interface{}) (interface{}, error) {
+	return &pvtRWSetMap{m: map[string]string{}}, nil
+}
+
+// pvtRWSetMap represents a cached element in store.cache
+type pvtRWSetMap struct {
+	mu sync.RWMutex
+	m  map[string]string
+}
+
+//func (p *pvtRWSetMap) get(k string) string {
+//	p.mu.RLock()
+//	defer p.mu.RUnlock()
+//
+//	return p.m[k]
+//}
+
+func (p *pvtRWSetMap) set(k string, v string) {
+	p.mu.Lock()
+	defer p.mu.Unlock()
+
+	p.m[k] = v
+}
+
+func (p *pvtRWSetMap) length() int {
+	p.mu.RLock()
+	defer p.mu.RUnlock()
+
+	return len(p.m)
+}
+
+func (p *pvtRWSetMap) delete(k string) {
+	p.mu.Lock()
+	defer p.mu.Unlock()
+
+	delete(p.m, k)
+}
+
+func (p *pvtRWSetMap) keys() []string {
+	p.mu.RLock()
+	defer p.mu.RUnlock()
+
+	keys := make([]string, 0, len(p.m))
+	for k := range p.m {
+		keys = append(keys, k)
+	}
+	return keys
+}
+
+// loadBlockHeight is a loader function of store.blockHeightCache
+func loadBlockHeight(key interface{}) (interface{}, error) {
+	return &txidsSlice{m: []string{}}, nil
+}
+
+// txidsSlice represents a cached element in store.blockHeightCache
+type txidsSlice struct {
+	mu sync.RWMutex
+	m  []string
+}
+
+func (p *txidsSlice) add(v string) {
+	p.mu.Lock()
+	defer p.mu.Unlock()
+
+	p.m = append(p.m, v)
+	sort.Strings(p.m) // ensures txids are sorted to help sort.search call in findTxidEntryInSlice below
+}
+
+// findTxidEntryInSlice will search for a str in the txidsSlice
```

Comment:
replace:
// findTxidEntryInSlice will search for a str in the txidsSlice
with
// findTxidEntryInSlice will search **for txid in txidsSlice**

Reply:
done